src/languages/rust.js

```diff
@@ -193,9 +193,19 @@ export default function(hljs) {
       {
         className: 'string',
         variants: [
-          { begin: /b?r(#*)"(.|\n)*?"\1(?!#)/ },
-          { begin: /b?'\\?(x\w{2}|u\w{4}|U\w{8}|.)'/ }
-        ]
+          {
+            begin: /b?r(#*)"(.|\n)*?"\1(?!#)/,
+            contains: [
+              hljs.BACKSLASH_ESCAPE
+            ]
+          },
+          {
+            begin: /b?'\\?(x\w{2}|u\w{4}|U\w{8}|.)'/,
+            contains: [
+              hljs.BACKSLASH_ESCAPE
+            ]
+          },
+        ],
```

Comment:
This doesn't work... you'd need `end` as well and need to rewrite any specific internals (like the unicode matching) to be sub rules inside the `contains`.  And you can't use match history (`\1`) across begin/end combos... how many `#` characters are used in practice? is there a max limit?

Reply:
I am sorry, I can't understand you.
Do you mean removing this change avoiding complexity or another thing?

Reply:
Please add test cases for these and you will see that they do not work they way they are currently written.

Reply:
You are right, this change didn't work and the contains at this these variants.
I will remove it and commit changes again.
Thanks in advance